### ISSUE TITLE
An option to reuse provisioned subdirs

### ISSUE THREAD

**Issue by moikot** [kubernetes-sigs/nfs-subdir-external-provisioner#5]
Current identity for provisioned subir includes the persisted volume name like this:

```golang
pvName := strings.Join([]string{pvcNamespace, pvcName, options.PVName}, "-")
```
Would it be possible to include an option to omit options.PVName so that when I re-provision my cluster I could re-use the already existing subdirs as PVs?

**Comment by dbsanfte** [comment]
Yes, this feels like a big missing feature. My NFS server persists across the lifecycle of my Kubernetes clusters: the latter are sometimes destroyed and recreated. NFS keeps my file state. I want to reuse existing files for new clusters. 

Since the PVName ends up having a GUID in it, a brand new folder is created every time and it's simply unnecessary. 

Maybe I'm missing something but I don't see a way to specify the PVName of the PersistentVolume that is be created to back the PVC?

**Comment by kmova** [comment]
@moikot @dbsanfte, would the following pattern work. 

<custom-cluster-prefix>-<pvc-namespace>-<pvc-name>, where custom-cluster-prefix can be provided via a storage class parameter. 

This should avoid any conflicts with multiple clusters using the same pvc-namespace and pvc-name.

**Comment by dbsanfte** [comment]
That would be fine for me!

**Comment by moikot** [comment]
The latest change should work for me too.

**Comment by geoff-coppertop** [comment]
Was this integrated? it looks like the provisioner is still using the UID.

**Comment by kmova** [comment]
The above suggestion needs to be coded. Lost track of this, will pick it up shortly.

**Comment by sifotes** [comment]
This seems like such key behavior, I assumed it was the default. 
Currently my workaround is to keep the PVCs defined and re bind them every time the pod comes up but as mentioned, this won't work for new clusters / auto scaling clusters, or even just including the PVC definition in the workload's manifest as would be preferred.

**Comment by sifotes** [comment]
Yes, it seems like that. A quick test shows that this works for my use case. I think @moikot would agree this can be closed.

**Comment by devrck** [comment]
I can confirm it works for me if I specify the `pathPattern`.

```
helm install nfs-subdir-external-provisioner nfs-subdir-external-provisioner/nfs-subdir-external-provisioner \
    --set nfs.server=x.x.x.x \
    --set nfs.path=/exported/path \
    --set storageClass.pathPattern= "${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}"
```

```
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: pvc
  annotations:
    nfs.io/storage-path: "/var/www/html/uploads"
spec:
  storageClassName: nfs-client
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 1Gi
```

**Comment by willzhang** [comment]
> I can confirm it works for me if I specify the `pathPattern` and in the PVC the `nfs.io/storage-path`
> 
> ```
> helm install nfs-subdir-external-provisioner nfs-subdir-external-provisioner/nfs-subdir-external-provisioner \
>     --set nfs.server=x.x.x.x \
>     --set nfs.path=/exported/path \
>     --set storageClass.pathPattern= "${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}"
> ```
> 
> ```
> ---
> apiVersion: v1
> kind: PersistentVolumeClaim
> metadata:
>   name: pvc
>   annotations:
>     nfs.io/storage-path: "/var/www/html/uploads"
> spec:
>   storageClassName: nfs-client
>   accessModes:
>     - ReadWriteMany
>   resources:
>     requests:
>       storage: 1Gi
> ```

why i failed
```
[root@master ~]# helm install nfs-subdir-external-provisioner nfs-subdir-external-provisioner/nfs-subdir-external-provisioner \
>   --namespace=nfs-provisioner \
>   --create-namespace \
>   --set image.repository=willdockerhub/nfs-subdir-external-provisioner \
>   --set image.tag=v4.0.2 \
>   --set replicaCount=2 \
>   --set storageClass.name=nfs-client \
>   --set storageClass.defaultClass=true \
>   --set storageClass.pathPattern= "${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}" \
>   --set nfs.server=192.168.0.21 \
>   --set nfs.path=/data/nfs/k8s
-bash: ${.PVC.namespace}/${.PVC.annotations.nfs.io/storage-path}: bad substitution
```
